File: src/main/java/championship/model/Match.java
package championship.model;

public class Match {
    private Player player1, player2;
    private boolean isStarted, isClosed;
    private int score1, score2;

    public Match(Player player1, Player player2) {
        setPlayer(1, player1);
        setPlayer(2, player2);
        score1 = 0;
        score2 = 0;
        isStarted = false;
        isClosed = false;
    }

    public void updateScorePlayer1(int score) {
        setScore(1, score);
    }

    public void updateScorePlayer2(int score) {
        setScore(2, score);
    }

    public boolean isStarted() {
        return isStarted;
    }

    public void start() {
        if (isStarted || isClosed) {
            throw new ChampionshipException("already started or closed");
        }
        isStarted = true;
    }

    public boolean isClosed() {
        return isClosed;
    }

    public void close() {
        if (!isStarted) {
            throw new ChampionshipException("Match is not started");
        }
        isClosed = true;
    }

    public Player[] getTwoPlayers() {
        return new Player[]{player1, player2};
    }

    public boolean isDraw() {
        if (!isClosed) {
            throw new ChampionshipException("Match still open");
        }
        return score1 == score2;
    }

    public Player getWinner() {
        if (!isClosed) {
            throw new ChampionshipException("Match still open");
        }
        if (isDraw()) {
            throw new ChampionshipException("No winner");
        }
        if (score1 >= score2) {
            return player1;
        } else {
            return player2;
        }
    }

    private void setScore(int number, int score) {
        if (!isStarted) {
            throw new ChampionshipException("match not started");
        }
        if (isClosed) {
            throw new ChampionshipException("match is closed");
        }
        if (score < 0) {
            throw new IllegalArgumentException("Score must be greater than 0");
        }
        switch (number) {
            case 1 : score1 = score;
                    break;
            case 2 : score2 = score;
                    break;
            default:
        }
    }

    private void setPlayer(int number, Player player) {
        if (player == null) {
            String msg = "player" + number + " cannot be null";
            throw new IllegalArgumentException(msg);
        }
        switch (number) {
            case 1 : player1 = player;
                    break;
            case 2 : player2 = player;
                    break;
            default:
        }
    }
}

File: csharp/championship/Match.cs
using System;

namespace championship
{
    public class Match
    {
        private Player _player1, _player2;

        private bool _started, _closed;

        private int _score1, _score2;

        public Match(Player player1, Player player2) {
            if (player1 == null) 
            {
                throw new ArgumentNullException("player1 is null");
            }

            if (player2 == null) 
            {
                throw new ArgumentNullException("player2 is null");
            }

            _player1 = player1;
            _player2 = player2;

            _started = false;
            _closed = false;
        }

        public bool Started
        {
            get
            {
                return _started;
            }
        }

        public bool Closed
        {
            get 
            {
                return _closed;
            }
        }

        public Player[] TwoPlayers
        {
            get
            {
                return new Player[2] {_player1, _player2};
            }
        }

        public void Start()
        {
            if (_started) 
            {
                throw new ChampionshipException("Already started !");
            }
            _started = true;
        }

        public void Close()
        {
            if (! _started) 
            {
                throw new ChampionshipException("Can't close unstarted match !");
            }
            if (_closed) 
            {
                throw new ChampionshipException("Match already closed !");
            }
            _closed = true;
        }

        public bool IsDraw() 
        {
            if (! _closed) 
            {
                throw new ChampionshipException("Match Still Open");
            }
            return _score1 == _score2;
        }

        public void UpdateScorePlayer1(int score) 
        {
            setScore(1, score);
        }

        public void UpdateScorePlayer2(int score) 
        {
            setScore(2, score);
        }

        public Player Winner 
        {
            get
            {
                if (!_closed) 
                {
                    throw new ChampionshipException("Match still open");
                }
                if (IsDraw()) 
                {
                    throw new ChampionshipException("No winner");
                }
                if (_score1 >= _score2) 
                {
                    return _player1;
                } 
                else 
                {
                    return _player2;
                }
            }
        }

        private void setScore(int number, int score) 
        {
            if (!_started) 
            {
                throw new ChampionshipException("match not started");
            }
            if (_closed) 
            {
                throw new ChampionshipException("match is closed");
            }
            if (score < 0) 
            {
                throw new ChampionshipException("Score must be greater than 0");
            }
            switch (number) {
                case 1 : _score1 = score;
                        break;
                case 2 : _score2 = score;
                        break;
            }
        }

    }
}
